### ISSUE TITLE
Predicates with comparisons don’t find all results.

### ISSUE THREAD

**Issue by simonjwright** [simonjwright/xia#2]
Predicates that involve comparisons against attributes only look at the first attribute.

If I have
```
   <class name="first_is_alpha">
     <stereotype name="alpha"/>
     <stereotype name="beta"/>
   </class>
```
then the query `class[stereotype/@name='beta']` fails to match, and the query `class[not(stereotype/@name='beta')]` succeeds.

Saxon 6.5.5 and the xsltproc provided with macOS agree that these are wrong.

**Comment by simonjwright** [comment]
It’s not just attributes: elements also fail.

**Comment by simonjwright** [comment]
Fixed in edc4782.
